File: OCH_Tao/ABC/A/245.js
const [A, B, C, D] = ((require("fs").readFileSync("/dev/stdin", "utf8").trim().split("\n"))[0].split(" ")).map(x => parseInt(x, 10));
let flag;
if (A < C) {
  flag = true;
} else if (C < A) {
  flag = false;
} else {
  if (B <= D) {
    flag = true;
  } else {
    flag = false;
  }
}
if (flag) {
  console.log("Takahashi");
} else {
  console.log("Aoki");
}

File: OCH_Tao/ABC/A/245.py
A,B,C,D = map(int,input().split())
if A<C:
  flag = True
elif C<A:
  flag = False
else:
  if B<D:
    flag = True
  elif D<B:
    flag = False
  else:
    flag = True
if flag:
  print("Takahashi")
else:
  print("Aoki")